src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -64,4 +65,18 @@ public void testParse() throws Exception {
         assertEquals(1, changes.size());
         assertEquals("HelloWorld", changes.iterator().next().getPath());
     }
+
+    @Test
+    public void testParseDeleteChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.addAnEmptyFileAndCommit(repository, "FileToDelete");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.removeFileAndCommit(repository, "FileToDelete");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be no changes detected", 0, changes.size());
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
```

Comment:
You will have to think your test pattern than.
If anything throws an exception before it, delete will never happen and repository will be left on disk.
If you drop the field and don't have a before/after, than this should be in a `finally` which means you'll have to move `repository` into the correct scope.